packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -182,6 +182,14 @@ export function updateWrapper(element: Element, props: Object) {
     } else if (node.value !== '' + value) {
       node.value = '' + value;
     }
+  } else if (
+    (props.type === 'submit' || props.type === 'reset') &&
```

Comment:
Nit: since we access `props.type` so much can you please read `type` in a variable early and use that